### PR TITLE
Ports @aledsage's Troubleshooting blog post to docs

### PR DESCRIPTION
**Pull request by nakomis**

### CODE REVIEW

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+## VM reachable
+If the VM is supposed to be accessible directly (e.g. from the public internet, or if in a private network
+then from a jump host)...
+
+### ping
+Can you `ping` the VM from the machine you are trying to reach it from.
+
+However, ping is over ICMP. If the VM is unreachable, it could be that the firewall forbids ICMP but still
+lets TCP traffic through).
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: Superfluous bracket.

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+This blog post gives some general tips for debugging connectivity issues, which are applicable to a 
+range of different service types. Choose those that are appropriate for your use-case.
+
+## VM reachable
+If the VM is supposed to be accessible directly (e.g. from the public internet, or if in a private network
+then from a jump host)...
+
+### ping
+Can you `ping` the VM from the machine you are trying to reach it from.
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: A number of these paragraphs should end with question marks.

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+### DNS and routing
+If using a hostname rather than IP, then is it resolving to a sensible IP?
+
+Is the route to the server sensible? (e.g. one can hit problems with proxy servers in a corporate
+network, or ISPs returning a default result for unknown hosts).
+
+The following commands can be useful:
+
+* `host` is a DNS lookup utility. e.g. `host www.google.com`.
+* `dig` stands for “domain information groper”. e.g. `dig www.google.com`.
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: These inverted commas are weird characters.

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+* `traceroute` prints the route that packets take to a network host. e.g. `traceroute www.google.com`.
+
+## Service is listening
+
+### Service responds
+Try connecting to the service from the VM iteslf. For example, `curl http://localhost:8080` for a
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: `iteslf`

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+web-service.
+
+On dev/test VMs, don’t be afraid to install the utilities you need such as `curl`, `telnet`, `nc`,
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: Weird apostrophe character.

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+For `netstat`, run with `sudo` to see the pid for all listed ports.
+
+## Firewalls
+On Linux, check if `iptables` is preventing the remote connection. On Windows, check the Windows Firewall.
+
+If it is acceptable (e.g. it is not a server in production), try turning off the firewall temporarily,
+and testing connectivity again. Remember to re-enable it afterwards! On CentOS, this is `sudo service
+iptables stop`. On Ubuntu, use `sudo ufw disable`. On Windows, go to `Start` -> `Control Panel` ->
+`Windows Firewall`, and use the “Turn off Windows Firewall”.
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: the "Turn off windows Firewall" what?

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+## Cloud firewalls
+Some clouds offer a firewall service, where ports need to be explicitly listed to be reachable.
+
+For example, (security groups for EC2-classic)
+[http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-network-security.html#ec2-classic-security-groups]
+have rules for the protocols and ports to be reachable from specific CIDRs.
+
+Check these settings via the cloud provider’s web-console (or API).
+Quick test of a listener port
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: Should this sentence be a heading?

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+which will be written out by the `nc -l` process before exiting.
+
+Similarly for UDP, you use `-lU`.
+
+You may first have to install `nc`, e.g. with `sudo yum install -y nc` or `sudo apt-get install netcat`.
+
+### Cloud load balancers
+For some use-cases, it is good practice to use the load balancer service offered by the cloud provider
+(e.g. (ELB in AWS)[http://aws.amazon.com/elasticloadbalancing/] or the (Cloudstack Load Balancer)
+[http://cloudstack.apache.org/docs/en-US/Apache_CloudStack/4.1.1/html/Installation_Guide/load-balancer-rules.html].
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: This link 404s

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+| nc <hostname> 8080`. If all works well, this will send “hello from client” over the TCP port 8080,
+which will be written out by the `nc -l` process before exiting.
+
+Similarly for UDP, you use `-lU`.
+
+You may first have to install `nc`, e.g. with `sudo yum install -y nc` or `sudo apt-get install netcat`.
+
+### Cloud load balancers
+For some use-cases, it is good practice to use the load balancer service offered by the cloud provider
+(e.g. (ELB in AWS)[http://aws.amazon.com/elasticloadbalancing/] or the (Cloudstack Load Balancer)
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: This bracket is unclosed.

File: docs/guide/dev/tips/troubleshooting-connectivity.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+the Cloud Provider’s web-console.
+
+### Guest wifi
+It is common for guest wifi to restrict access to only specific ports (e.g. 80 and 443, restricting
+ssh over port 22 etc).
+
+Normally your best bet is then to abandon the guest wifi (e.g. to tether to a mobile phone instead).
+
+There are some unconventional workarounds such as (configuring sshd to listen on port 80 so you can
+use an ssh tunnels) [http://askubuntu.com/questions/107173/is-it-possible-to-ssh-through-port-80].
```
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: I think these `(text)[link]` link combinations are all the wrong way round. The rest of the website docs use `[text](link)`.

File: docs/guide/dev/tips/troubleshooting-connectivity.md (hunk 3 above)
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: You could replace all of them with this in vim: `:%s/“\|”/"/g`. And for the weird apostrophes: `:%s/’/"/g`.

File: docs/guide/dev/tips/troubleshooting-connectivity.md (hunk 10 above)
Comment (sjcorbett) [inline docs/guide/dev/tips/troubleshooting-connectivity.md]: Superfluous space between brackets. Unsure if it matters.

**Comment by nakomis** [comment]
PR comments addressed

**Pull request merged**